minemeld/ft/taxii2.py

```diff
@@ -0,0 +1,634 @@
+#  Copyright 2015-present Palo Alto Networks, Inc
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import absolute_import
+
+import logging
+import os
+import yaml
+from datetime import datetime, timedelta
+from uuid import UUID
+
+import pytz
+import requests
+import requests.structures
+import ujson
+
+from stix2patterns.pattern import Pattern, ParseException
+
+from . import basepoller
+
+from .utils import dt_to_millisec, interval_in_sec
+
+LOG = logging.getLogger(__name__)
+
+_STIX2_MINEMELD_HASHES = [
+    'ssdeep',
+    'md5',
+    'sha1',
+    'sha256',
+    'sha512'
+]
+
+_STIX2_TYPES_TO_MM_TYPES = {
+    'ipv4-addr': 'IPv4',
+    'ipv6-addr': 'IPv6',
+    'domain': 'domain',
+    'domain-name': 'domain',
+    'url': 'URL',
+    'file': None,
+    'md5': 'md5',
+    'sha-1': 'sha1',
+    'sha-256': 'sha256'
+}
+
+
+class Taxii2Client(basepoller.BasePollerFT):
+    def __init__(self, name, chassis, config):
+        self.poll_service = None
+        self.last_taxii2_run = None
+        self.last_stix2_package_ts = None
+
+        super(Taxii2Client, self).__init__(name, chassis, config)
+
+    def configure(self):
+        super(Taxii2Client, self).configure()
+
+        self.initial_interval = self.config.get('initial_interval', '1d')
+        self.initial_interval = interval_in_sec(self.initial_interval)
+        if self.initial_interval is None:
+            LOG.error('%s - wrong initial_interval format: %s', self.name, self.initial_interval)
+            self.initial_interval = 86400
+        self.max_poll_dt = self.config.get('max_poll_dt', 86400)
+
+        # options for processing
+        self.ip_version_auto_detect = self.config.get('ip_version_auto_detect', True)
```

Comment:
It seems that `ip_version_auto_detect`, `ignore_composition_operator`, `create_fake_indicator`, `create_fake_indicator` are no longer needed. Any reason to keep them in the config?

Reply:
Removed.